pulsar-functions/runtime/src/main/java/org/apache/pulsar/functions/runtime/kubernetes/KubernetesRuntime.java

```diff
@@ -200,7 +204,11 @@
         this.pulsarRootDir = pulsarRootDir;
         this.configAdminCLI = configAdminCLI;
         this.userCodePkgUrl = userCodePkgUrl;
-        this.originalCodeFileName = pulsarRootDir + "/" + originalCodeFileName;
+        this.downloadDirectory = this.pulsarRootDir; // for backward comp
+        if (StringUtils.isNotBlank(downloadDirectory)) {
+            this.downloadDirectory = downloadDirectory;
+        }
```

Comment:
how about `this.downloadDirectory = StringUtils.isNotBlank(downloadDirectory)?downloadDirectory:this.pulsarRootDir;` to avoid some confusion reading this part of code?